src/components/application_manager/test/commands/mobile/perform_audio_pass_thru_test.cc

```diff
@@ -108,104 +125,503 @@ class PerformAudioPassThruRequestTest
+  utils::SharedPtr<PerformAudioPassThruRequest> command =
+      CreateCommand<PerformAudioPassThruRequest>(msg_mobile);
+
+  MessageSharedPtr msg_tts =
+      CreateHMIMessageSO(hmi_apis::Common_Result::SUCCESS, kCmdId);
+  am::event_engine::Event event_tts(hmi_apis::FunctionID::TTS_Speak);
+  event_tts.set_smart_object(*msg_tts);
 
   ON_CALL(hmi_interfaces_,
           GetInterfaceState(am::HmiInterfaces::HMI_INTERFACE_UI))
-      .WillByDefault(Return(am::HmiInterfaces::STATE_NOT_AVAILABLE));
+      .WillByDefault(Return(am::HmiInterfaces::STATE_AVAILABLE));
   ON_CALL(hmi_interfaces_,
           GetInterfaceState(am::HmiInterfaces::HMI_INTERFACE_TTS))
-      .WillByDefault(Return(am::HmiInterfaces::STATE_NOT_AVAILABLE));
+      .WillByDefault(Return(am::HmiInterfaces::STATE_AVAILABLE));
+  EXPECT_CALL(mock_message_helper_,
+              HMIToMobileResult(hmi_apis::Common_Result::SUCCESS))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
 
-  MessageSharedPtr response_msg_tts =
-      CreateMessage(smart_objects::SmartType_Map);
-  (*response_msg_tts)[am::strings::params][am::hmi_response::code] =
-      hmi_apis::Common_Result::SUCCESS;
-  (*response_msg_tts)[am::strings::msg_params][am::strings::cmd_id] = kCmdId;
+  command->on_event(event_tts);
+
+  MessageSharedPtr msg_ui =
+      CreateHMIMessageSO(hmi_apis::Common_Result::TRUNCATED_DATA, kCommandId);
+  Event event_ui(hmi_apis::FunctionID::UI_PerformAudioPassThru);
+  event_ui.set_smart_object(*msg_ui);
+
+  EXPECT_CALL(mock_message_helper_,
+              HMIToMobileResult(hmi_apis::Common_Result::TRUNCATED_DATA))
+      .WillOnce(Return(mobile_apis::Result::TRUNCATED_DATA));
+  EXPECT_CALL(app_mngr_, EndAudioPassThrough()).WillOnce(Return(false));
+
+  MessageSharedPtr msg_mobile_response;
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(_, am::commands::Command::CommandOrigin::ORIGIN_SDL))
+      .WillOnce(DoAll(SaveArg<0>(&msg_mobile_response), Return(true)));
+
+  command->on_event(event_ui);
+
+  ResultCommandCodeExpectations(
+      msg_mobile_response, am::mobile_api::Result::TRUNCATED_DATA, false);
+}
+
+TEST_F(PerformAudioPassThruRequestTest,
+       OnEvent_TTSHmiSendTruncatedData_TRUNCATED_DATA) {
+  MessageSharedPtr msg_mobile = CreateMobileMessageSO();
+  utils::SharedPtr<PerformAudioPassThruRequest> command =
+      CreateCommand<PerformAudioPassThruRequest>(msg_mobile);
+
+  MessageSharedPtr msg_tts =
+      CreateHMIMessageSO(hmi_apis::Common_Result::TRUNCATED_DATA, kCmdId);
+  am::event_engine::Event event_tts(hmi_apis::FunctionID::TTS_Speak);
+  event_tts.set_smart_object(*msg_tts);
+
+  ON_CALL(hmi_interfaces_,
+          GetInterfaceState(am::HmiInterfaces::HMI_INTERFACE_UI))
+      .WillByDefault(Return(am::HmiInterfaces::STATE_AVAILABLE));
+  ON_CALL(hmi_interfaces_,
+          GetInterfaceState(am::HmiInterfaces::HMI_INTERFACE_TTS))
+      .WillByDefault(Return(am::HmiInterfaces::STATE_AVAILABLE));
+  EXPECT_CALL(mock_message_helper_,
```

Comment:
@tkireva This defines the expectation behavior. However is there an expectation for these calls or there are notification for unexpected mock calls? Here and in the other places.